app/plugin_requests.go

```diff
@@ -38,22 +40,42 @@ func (a *App) ServePluginRequest(w http.ResponseWriter, r *http.Request) {
 
 func (a *App) servePluginRequest(w http.ResponseWriter, r *http.Request, handler func(*plugin.Context, http.ResponseWriter, *http.Request)) {
 	token := ""
+	context := &plugin.Context{}
 
 	authHeader := r.Header.Get(model.HEADER_AUTH)
 	if strings.HasPrefix(strings.ToUpper(authHeader), model.HEADER_BEARER+" ") {
```

Comment:
This block of code interests me. I'm not sure why I missed it in the first round of reviews, but just some clarifying questions:
* Will I now be prohibited from making authenticated requests through a plugin's API unless I manually send the `HEADER_REQUESTED_WITH` header with the requisite value? I think the webapp does this automatically for XMLHttpRequests, but what if I'm using curl/[http](httpie.org) to hit the plugin's API, or integrating with another application that interfaces with the API? (This might be by design and/or a requirement of CSRF protection, but just want to make sure.)
* What if my plugin's API endpoint isn't parsing forms, but accepts JSON? How do I `POST` a JSON payload to my plugin endpoint /and/ set the `csrf` token in the body as a form value /and/ still have an authenticated request? I guess this boils down to the previous question requiring me to set this special header? Is it a standard header, or something custom we've adopted?
* It looks like we support getting an `access_token` from the query string. Ignoring any security issues with that, I guess this happens to work because we don't issue access tokens corresponding csrf tokens. Of course, someone could pass their current session's token /as/ an access token of sorts, but I suppose we don't support that.

Reply:
* If you don't have the XMLHttpRequest header set you will be required to send a CSRF Token, that should _not_ be the target when cookie authentication is used and was planned for a migration period, since that was the previous behavior. In the long-term I'd like to enforce CSRF tokens for all requests that were made with cookie auth.
* You can send a csrf token via GET parameters as well, we could add a header option optionally. The url query parameter is supported ootb by `FormValue` in Go, the header would be an additional code enhancement.
* That code snippet isn't from me, but according to my understanding `access_token == session token != csrf token` - Hope that helps?

*Important:* I added a new check that the CSRF enforcement is only applied for cookie authentication after our chat in pre-release. FYI @crspeller